rm/src/lib.rs

```diff
@@ -108,64 +110,70 @@ impl Args for Options {
-
-            if project::get(&storage, urn)?.is_none() {
-                anyhow::bail!("project {} does not exist", &urn);
-            }
-            let namespace = monorepo
-                .join("refs")
-                .join("namespaces")
-                .join(&urn.encode_id());
-
-            if !options.confirm
-                || term::confirm(format!(
-                    "Are you sure you would like to delete {}?",
-                    term::format::dim(namespace.display())
-                ))
-            {
-                rad_untrack::execute(urn, None, rad_untrack::Options { peer: None }, &profile)?;
-                fs::remove_dir_all(namespace)?;
-                term::success!("Successfully removed project {}", &urn);
-            }
-        }
-        Object::Profile(id) => {
-            let profile = ctx.profile()?;
-            if profile.id() != id {
-                let profile = profile::get(id)?;
-                let read_only = profile::read_only(&profile)?;
-                let config = read_only.config()?;
-                let username = config.user_name()?;
-
-                if options.confirm
-                    && !term::confirm(format!(
-                        "Are you sure you would like to delete {} ({})?",
-                        term::format::dim(id),
-                        term::format::dim(username)
+            if let Ok(Some(_)) = project::get(&storage, urn) {
+                let monorepo = profile.paths().git_dir();
+                let namespace = monorepo
+                    .join("refs")
+                    .join("namespaces")
+                    .join(&urn.encode_id());
+
+                if !options.confirm
+                    || term::confirm(format!(
+                        "Are you sure you would like to delete {}?",
+                        term::format::dim(namespace.display())
                     ))
                 {
-                    return Ok(());
+                    rad_untrack::execute(urn, None, rad_untrack::Options { peer: None }, &profile)?;
+                    fs::remove_dir_all(namespace)?;
+                    term::success!("Successfully removed project {}", &urn);
                 }
-
-                if options.passphrase {
-                    let passphrase = term::read_passphrase(options.stdin, false)?;
-                    if keys::load_secret_key(&profile, passphrase).is_err() {
-                        anyhow::bail!(format!("Invalid passphrase supplied."));
+            } else {
+                anyhow::bail!("project {} does not exist", &urn)
+            }
+        }
+        Object::User(peer_id) => {
+            let mut profiles = profile::list()?;
+            if storage.peer_id() != peer_id {
+                profiles.retain(|p| match profile::read_only(p) {
+                    Ok(s) => s.peer_id() == peer_id,
+                    Err(_) => false,
+                });
+
+                if let Some(other) = profiles.last() {
```

Comment:
I think you could combine `retain` + `last` by doing:
```rust
if let Some(other) = profiles.find(|p| match profile::read_only(p) {
  Ok(s) => s.peer_id() == peer_id,
  Err(_) => false,
}) => ...
```

Reply:
Yes, also you could use `map` to not have to re-fetch the `profile::read_only` below